src/main/java/com/jakewharton/retrofit2/converter/kotlinx/serialization/Serializer.kt

```diff
@@ -8,6 +8,7 @@ import kotlinx.serialization.KSerializer
 import kotlinx.serialization.SerialFormat
 import kotlinx.serialization.SerializationStrategy
 import kotlinx.serialization.StringFormat
+import kotlinx.serialization.json.decodeFromStream
```

Comment:
Does this only work for JSON? We are format agnostic and cannot limit the usage to a single format.

Reply:
Right now, yes, I think.

```

/**
 * Deserializes the contents of given [stream] to to the value of type [T] using UTF-8 encoding and
 * deserializer retrieved from the reified type parameter.
 *
 * Note that this functions expects that exactly one object would be present in the stream
 * and throws an exception if there are any dangling bytes after an object.
 *
 * @throws [SerializationException] if the given JSON input cannot be deserialized to the value of type [T].
 * @throws [IOException] If an I/O error occurs and stream can't be read from.
 */
@ExperimentalSerializationApi
public inline fun <reified T> Json.decodeFromStream(stream: InputStream): T =
    decodeFromStream(serializersModule.serializer(), stream)
```

Reply:
Okay so we can't use it then, and even if it were available for all formats we cannot use it until it's stable.